pkg/tbtc/node.go

```diff
@@ -173,17 +176,77 @@ func (n *node) joinDKGIfEligible(seed *big.Int, startBlockNumber uint64) {
 					return
 				}
 
+				publicationStartBlock := endBlock
+				operatingMemberIDs := result.Group.OperatingMemberIDs()
+				dkgResultChannel := make(chan *DKGResultSubmittedEvent)
+
+				dkgResultSubscription := n.chain.OnDKGResultSubmitted(
+					func(event *DKGResultSubmittedEvent) {
+						dkgResultChannel <- event
+					},
+				)
+				defer dkgResultSubscription.Unsubscribe()
+
+				err = dkg.Publish(
+					logger,
+					seed,
+					publicationStartBlock,
+					memberIndex,
+					blockCounter,
+					broadcastChannel,
+					membershipValidator,
+					n.chain,
+					n.chain,
+					result,
+				)
+				if err != nil {
+					// Result publication failed. It means that either the result this
+					// member proposed is not supported by the majority of group members or
+					// that the chain interaction failed. In either case, we observe the
+					// chain for the result published by any other group member and based
+					// on that, we decide whether we should stay in the final group
+					// or drop our membership.
+					logger.Warningf(
+						"[member:%v] DKG result publication process failed [%v]",
+						memberIndex,
+						err,
+					)
+
+					if operatingMemberIDs, err = n.decideMemberFate(
+						memberIndex,
+						dkgResultChannel,
+						publicationStartBlock,
+						result,
+					); err != nil {
+						logger.Errorf(
+							"failed to handle DKG result publishing failure: [%v]",
+							err,
+						)
+						return
+					}
+				}
+
+				signingGroupOperators, err := n.resolveSigningGroupOperators(
+					selectedSigningGroupOperators,
+					operatingMemberIDs,
+				)
+				if err != nil {
+					logger.Errorf(
+						"failed to resolve group operators: [%v]",
+						err,
+					)
+					return
+				}
+
 				// TODO: Snapshot the key material before doing on-chain result
 				//       submission.
 
-				// TODO: Submit the result using the chain layer.
-
 				// TODO: The final `signingGroupOperators` may differ from
 				//       the original `selectedSigningGroupOperators`.
 				//       Consider that when integrating the retry algorithm.
 				signer := newSigner(
```

Comment:
Shouldn't we remove the above TODO as well?

Reply:
Done in 4c76a72cb145b310ff8710fa1b778d0382716ac9.